core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -176,6 +176,17 @@ abstract class TaskContext extends Serializable {
    */
   def getLocalProperty(key: String): String
 
+  /**
+   * The indeterminate attempt ID of the stage that this task belongs to, it returns the stage
+   * attempt number while the stage is not determinate and returns none on the contrary.
+   */
+  def indeterminateStageAttemptId(shuffleId: Int): Option[Int] = {
+    val id = getLocalProperty(SparkContext.INDETERMINATE_STAGE_ATTEMPT_ID_PREFIX + shuffleId)
```

Comment:
do we need to include shuffle id?

Reply:
I think we should include shuffle id here, for an indeterminate shuffle map task which parent stage also indeterminate, we both need parent stage id and current id here for shuffle reader and writer. Also for the more complex scenario, for join/cogroup operation, maybe we have more indeterminate parents for single shuffle map stage.

Reply:
makes sense